clients/index.js

```diff
@@ -300,6 +319,34 @@ function setupChannel(cb) {
     }
 
     self._controlServer.listen(self._controlPort, listenReady.signal);
+
+    function getHostForTChannel() {
+        var host = localIp();
+        var reason = validateHost(host);
+        if (reason !== null) {
+            nextAttempt(host, reason);
+            return;
+        }
+        self.tchannel.listen(self._port, host);
+    }
+
+    function nextAttempt(host, reason) {
+        var delay = self._getHostForTchannelAttemptBackoff * Math.pow(2, getHostAttemptCount);
```

Comment:
We do rolling deploys and have a 2minute timeout on a worker coming up.

I'd be nice to have it bail out in <2minute.